kernel/src/main/scala/org/typelevel/sbt/kernel/V.scala

```diff
@@ -0,0 +1,63 @@
+package org.typelevel.sbt.kernel
+
+import scala.util.Try
+
+final case class V(
+    major: Int,
+    minor: Int,
+    patch: Option[Int],
+    prerelease: Option[String]
+) extends Ordered[V] {
+
+  override def toString: String =
+    s"$major.$minor${patch.fold("")(p => s".$p")}${prerelease.fold("")(p => s"-$p")}"
+
+  def isPrerelease: Boolean = prerelease.nonEmpty
+
+  def isSameSeries(that: V): Boolean =
+    this.major == that.major && this.minor == that.minor
+
+  def mustBeBinCompatWith(that: V): Boolean =
+    this >= that && !that.isPrerelease && this.major == that.major && (major > 0 || this.minor == that.minor)
+
+  def compare(that: V): Int = {
+    val x = this.major.compare(that.major)
+    if (x != 0) return x
+    val y = this.minor.compare(that.minor)
+    if (y != 0) return y
+    (this.patch, that.patch) match {
+      case (None, None) => 0
+      case (None, Some(patch)) => 1
+      case (Some(patch), None) => -1
+      case (Some(thisPatch), Some(thatPatch)) =>
+        val z = thisPatch.compare(thatPatch)
+        if (z != 0) return z
+        (this.prerelease, that.prerelease) match {
+          case (None, None) => 0
+          case (Some(_), None) => 1
+          case (None, Some(_)) => -1
+          case (Some(thisPrerelease), Some(thatPrerelease)) =>
```

Comment:
`M9` vs. `M10`?  Splitting on word-number boundaries would do the right thing.

Reply:
Thanks, this can definitely be improved.

Reply:
If you don't mind, can we shunt this into a follow-up PR? Ordering of pre-releases is not critical for MiMa, since pre-releases don't impact bincompat. Doing this right means accounting for `x.y.z-RCn` comes _before_ `x.y.z` but `x.y.z-c6d2fdf` comes _after_ `x.y.z`. So for this to actually work we'll have to make some assumptions about how people version their stuff.

Reply:
:+1: #14